### PR TITLE
helm: ability to parameterize the number replicas MIC deployment

### PR DESCRIPTION
**Pull request by blafry**
**Reason for Change**:

Possibility to control the number of MIC replicas.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [ ] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [ ] ran `make precommit`

**Issue Fixed**:

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [x] no

**Notes for Reviewers**:

### CODE REVIEW

**Review by chewong — COMMENTED** [review]
Thanks for opening the PR. Could you make the changes in `manifest_staging` folder instead?

**Comment by aramase** [comment]
@blafry What is the reason for having more than than 2 replicas for MIC?

**Comment by blafry** [comment]
> Thanks for opening the PR. Could you make the changes in `manifest_staging` folder instead?
@chewong of course, done.

> @blafry What is the reason for having more than than 2 replicas for MIC?
@aramase the value of the number of replicas was transferred from the deployment, I did not change the author's intention.

**Comment by aramase** [comment]
> @aramase the value of the number of replicas was transferred from the deployment, I did not change the author's intention.

@blafry I'm curious what is the motivation for making this configurable? We set the replicas to 2 today because one is the leader and other is backup.

**Comment by blafry** [comment]
> > @aramase the value of the number of replicas was transferred from the deployment, I did not change the author's intention.
> 
> @blafry I'm curious what is the motivation for making this configurable? We set the replicas to 2 today because one is the leader and other is backup.

In non-production setups or where the services requiring authorization are not critical, a scale of 1 MIC will be sufficient. So it would be convenient to have it configurable.

**Comment by aramase** [comment]
> @aramase in non-production setups or where the services requiring authorization are not critical, a scale of 1 MIC will be sufficient. So it would be convenient to have it configurable.

Thanks for the clarification.

File: manifest_staging/charts/aad-pod-identity/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 description: Deploy components for aad-pod-identity
 name: aad-pod-identity
-version: 4.0.0
+version: 4.1.0
```
Comment (aramase) [inline manifest_staging/charts/aad-pod-identity/Chart.yaml:4]: nit: this change isn't required as we update the version before the release but not a blocker for this PR

**Review by aramase — APPROVED** [review]

**Review by chewong — APPROVED** [review]

**Pull request merged**